### PR TITLE
feat: simulator linux support

### LINKED ISSUE #1108: feat: simulator linux support
Add support for building and running the simulator on Linux

### PR DESCRIPTION
**Pull request by fmorency**
## Explanation

Closes #1108 

This PR adds support for building the simulator on Linux.

Note: Building the simulator with `-e dev` environment on a freshly cloned repository doesn't work. I'm not sure the error is related to this PR. It fails with
```shell
[100%] Linking C executable simulator
/usr/bin/ld: CMakeFiles/simulator.dir/src/ui/gui_assets/wallet/walletXrptoolkit.c.obj:(.rodata+0x0): multiple definition of `walletXRPToolkit_map'; CMakeFiles/simulator.dir/src/ui/gui_assets/wallet/walletXRPToolkit.c.obj:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/simulator.dir/src/ui/gui_assets/wallet/walletXrptoolkit.c.obj:(.data.rel.ro.local+0x0): multiple definition of `walletXRPToolkit'; CMakeFiles/simulator.dir/src/ui/gui_assets/wallet/walletXRPToolkit.c.obj:(.data.rel.ro.local+0x0): first defined here
/usr/bin/ld: CMakeFiles/simulator.dir/src/ui/gui_assets/walletList/walletListBlockwallet.c.obj:(.rodata+0x0): multiple definition of `walletListBlockWallet_map'; CMakeFiles/simulator.dir/src/ui/gui_assets/walletList/walletListBlockWallet.c.obj:(.rodata+0x0): first defined here
/usr/bin/ld: CMakeFiles/simulator.dir/src/ui/gui_assets/walletList/walletListBlockwallet.c.obj:(.data.rel.ro.local+0x0): multiple definition of `walletListBlockWallet'; CMakeFiles/simulator.dir/src/ui/gui_assets/walletList/walletListBlockWallet.c.obj:(.data.rel.ro.local+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/simulator.dir/build.make:10882: simulator] Error 1
make[1]: *** [CMakeFiles/Makefile2:115: CMakeFiles/simulator.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```
Note: I don't have access to a Windows/MacOS computer; I can't test if something broke.

## Pre-merge check list
- [x] PR run build successfully on local machine.
- [ ] All unit tests passed locally.

## How to test

Run
```shell
python3 build.py -o simulator -e production
```
on a Linux computer with the necessary dependencies installed, e.g., cbindgen, sdl2, ...

When the build finishes, run
```shell
./build/simulator
```

### CODE REVIEW

**Comment by soralit** [comment]
Thanks for PR. The problem may be related to the compiler you are using. You can change your compiler to see what will happen.

**Comment by fmorency** [comment]
> Thanks for PR. The problem may be related to the compiler you are using. You can change your compiler to see what will happen.

Thank you for the suggestion.

I get the same error with Clang+LLD, Clang+Mold, and GCC14+LD.

I noticed the files causing the error against the `dev` environment are listed differently in `git status`.



Again, I don't see this error when using the `production` environment.

**Comment by fmorency** [comment]
@soralit I fixed the build issue.

**Comment by 8de2fdb0** [comment]
@fmorency 

I tried your patch building inside the `Dockerfile` from this repo,  I added libsdl2 and cbindgen via apt.

Somehow the the linker flag `-m`  is missing, but I can't figure out where to add it.

```
[100%] Linking C executable simulator
/usr/bin/ld: ../ui_simulator/lib/rust-builds/librust_c.a(time-b84229e30c1d67d4.time.c1da2c4af1189564-cgu.15.rcgu.o): undefined reference to symbol 'fmod@@GLIBC_2.2.5'
/usr/bin/ld: /lib/x86_64-linux-gnu/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/simulator.dir/build.make:10285: simulator] Error 1
make[1]: *** [CMakeFiles/Makefile2:114: CMakeFiles/simulator.dir/all] Error 2
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -341,6 +341,9 @@ else()
         ${CRYPTO_SLIP39}
     )
     add_dependencies(${PROJECT_NAME} rust_c)
+    if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
+      target_link_libraries(${PROJECT_NAME} PRIVATE m m)
+    endif()
     target_link_libraries(${PROJECT_NAME} PRIVATE SDL2 ${CMAKE_SOURCE_DIR}/ui_simulator/lib/rust-builds/librust_c.a)
```
Comment (8de2fdb0) [inline CMakeLists.txt:347]:
```suggestion
    target_link_libraries(${PROJECT_NAME} PRIVATE SDL2 ${CMAKE_SOURCE_DIR}/ui_simulator/lib/rust-builds/librust_c.a)
    if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
      target_link_libraries(${PROJECT_NAME} PRIVATE m dl pthread )
    endif()
```

This fixes a linker issue that I had when using the `Dokerfile` provided in the repo. 

As mentioned I added `libsdl2-dev` to the `apt-get install` in the Dockerfile.

**Comment by soralit** [comment]
I will do a simple test and merge this PR.

**Review by soralit — APPROVED** [review]

**Pull request merged**